### ISSUE TITLE
Add Oneshot mode to the LFO

### ISSUE THREAD

**Issue by mkruselj** [surge-synthesizer/surge#6692]
We definitely have enough room above the Unipolar button for a Oneshot button. It would pertain only to the LFO and not the EG - basically play one single LFO cycle then freeze the LFO output at the last played value. This is something that Shortcircuit's LFOs have, too, so it'd bring nice parity between the two.

**Comment by Andreya-Autumn** [comment]
Solid idea. If I understand that right it'd make the S&H LFO shape behave just like the Random modulator yeah? 

What'd the noise LFO shape do then? Random walk until reaching the time set by the rate knob I guess? That could be a really cool function, but to really reach full potential it would require a way to control the rate of the random walking separately from the amount of time before the "cycle" is over. Hmm...

**Comment by Andreya-Autumn** [comment]
And it only makes sense in keytrigger mode also right? At least I can't think what such a function would do in freerun mode?

So maybe simplest initial implementation is: The option is greyed out when selecting other combinations for which it makes no sense. Which might be freerun mode, possibly S&H because redundant, and Noise because requires parameter trickery. Aksi sequencer and formula because like, what?

**Comment by mkruselj** [comment]
Yeah S&H would behave like Random modulator. And it would basically behave the same in all the trigger modes, I don't think it needs to be grayed out in modes other than Keytrigger. Noise LFO would be like S&H but it would be smoothed between phase cycles I reckon (because that's all the Noise LFO is - a cubic interpolated S&H). Step sequencer would also be entirely valid - plays the sequence once then stops at last step - loop wouldn't be played. Formula is also valid if it generates a cyclical thing, I'd suspect.

**Comment by Andreya-Autumn** [comment]
Ah ok, that makes sense about the noise LFO and step seq shapes. 

I'm not sure I see the point in freerun mode though. Wouldn't that just mean it'd do one cycle when pressing play and then never again? That doesn't seem super useful? Or am I missing something?

**Comment by mkruselj** [comment]
I guess we can cross-reference the behavior with Shortcircuit, ultimately.
